Blog-projects/Widget-window-in-Qt-Quick-app/fontcontrolswidgetsform.cpp

```diff
@@ -0,0 +1,47 @@
+#include "fontcontrolswidgetsform.h"
+#include "ui_fontcontrolswidgetsform.h"
+
+FontControlsWidgetsForm::FontControlsWidgetsForm(QWidget *parent)
+    : QWidget(parent)
+    , ui(new Ui::FontControlsWidgetsForm)
+{
+    ui->setupUi(this);
+    connect(ui->lineEdit, &QLineEdit::textEdited, this, &FontControlsWidgetsForm::lineEdit_textChanged);
+    connect(ui->fontComboBox, &QFontComboBox::currentFontChanged, this, &FontControlsWidgetsForm::fontComboBox_currentFontChanged);
+    connect(ui->pushButton, &QPushButton::clicked, this, &FontControlsWidgetsForm::pushButton_clicked);
+}
+
+FontControlsWidgetsForm::~FontControlsWidgetsForm()
+{
+    delete ui;
+}
+
+const QString FontControlsWidgetsForm::text()
+{
+    return ui->lineEdit->text();
+}
+
+void FontControlsWidgetsForm::lineEdit_textChanged(const QString &arg1)
+{
+    emit textChanged();
+}
+
+void FontControlsWidgetsForm::setText(const QString &text)
+{
+    ui->lineEdit->setText(text);
+    emit textChanged();
+}
+
+const QString FontControlsWidgetsForm::font()
+{
+    return ui->fontComboBox->currentFont().family();
+}
+
+void FontControlsWidgetsForm::closeEvent(QCloseEvent *event) {
+    QApplication::quit();
+}
+
+void FontControlsWidgetsForm::fontComboBox_currentFontChanged(const QFont &f)
```

Comment:
OK I see why you asked about casing... This snake/camel mix is weird ;)

Suggestions:
`slotFontChanged`
`onFontChanged`

or in you really need to mention the originating widget, say if you have two lineEdits and you were connecting to `textChanged()` from each, you'd say something like `slotNameChanged` and `slotCountryChanged` or possibly `slotNameTextChanged`/`slotCountryTextChanged`. But it's not like the exact name of the signal has to be in the slot name. Only if you were connecting to multiple similar signals.